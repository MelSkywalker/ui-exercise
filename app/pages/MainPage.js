import React, { Fragment } from 'react';
import Header from '../components/Header';
import SideBar from '../components/SideBar';
import EmailList from '../components/EmailList';
import './mainPage.scss';

const MainPage = ({status}) => {
	console.log(status);
	return (
		<div className="main-page">
			<Header />
			{status === 'loaded' &&
				<div className="page-container">
					<Fragment>
						<SideBar />
						<EmailList />
					</Fragment>
				</div>
			}
		</div>
	)
}

export default MainPage;