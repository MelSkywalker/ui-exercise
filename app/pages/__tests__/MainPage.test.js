import React from 'react';
import { shallow } from 'enzyme';

import MainPage from '../MainPage';

describe('pages | MainPage', () => {
	it('it renders the component with data', () => {
		const component = shallow(<MainPage status={'loaded'} />);
		expect(component).toMatchSnapshot();
	});
	it('it renders the component without data', () => {
		const component = shallow(<MainPage status={'loading'} />);
		expect(component).toMatchSnapshot();
	});
});
