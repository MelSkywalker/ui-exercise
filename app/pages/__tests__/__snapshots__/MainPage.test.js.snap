// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages | MainPage it renders the component with data 1`] = `
<div
  className="main-page"
>
  <Header />
  <div
    className="page-container"
  >
    <SideBar />
    <EmailList />
  </div>
</div>
`;

exports[`pages | MainPage it renders the component without data 1`] = `
<div
  className="main-page"
>
  <Header />
</div>
`;
